### PR TITLE
Document the provider CIPHER operation

### PR DESCRIPTION
**Pull request by mattcaswell**
Extends the existing provider documentation with information about the
CIPHER operation. This is primarily for provider authors.

### CODE REVIEW

File: doc/man3/EVP_EncryptInit.pod [hunk 1]
```diff
@@ -278,8 +278,11 @@ an B<EVP_CIPHER_CTX> structure.
 EVP_CIPHER_mode() and EVP_CIPHER_CTX_mode() return the block cipher mode:
 EVP_CIPH_ECB_MODE, EVP_CIPH_CBC_MODE, EVP_CIPH_CFB_MODE, EVP_CIPH_OFB_MODE,
 EVP_CIPH_CTR_MODE, EVP_CIPH_GCM_MODE, EVP_CIPH_CCM_MODE, EVP_CIPH_XTS_MODE,
-EVP_CIPH_WRAP_MODE or EVP_CIPH_OCB_MODE. If the cipher is a stream cipher then
-EVP_CIPH_STREAM_CIPHER is returned.
+EVP_CIPH_WRAP_MODE, EVP_CIPH_OCB_MODE or EVP_CIPH_SIV_MODE. If the cipher is a
+stream cipher then EVP_CIPH_STREAM_CIPHER is returned.
+
+EVP_CIPHER_flags() return any flags associated with the cipher. See
```
Comment (paulidale) [inline doc/man3/EVP_EncryptInit.pod]: _returns_

File: doc/man7/provider-cipher.pod [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+OP_cipher_dupctx() should duplicate the provider side cipher context in the
+B<cctx> parameter and return the duplicate copy.
+
+=head2 Encryption/Decryption Functions
+
+OP_cipher_encrypt_init() initialises a cipher operation for encryption given a
+newly created provider side cipher context in the B<cctx> paramter.
+The key to be used is given in B<key> which is B<keylen> bytes long.
+The IV to be used is given in B<iv> which B<ivlen> bytes long.
```
Comment (paulidale) [inline doc/man7/provider-cipher.pod]: _which is_

File: doc/man7/provider-cipher.pod [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+OP_cipher_update() should encrypt/decrypt B<inl> bytes of data at the location
+pointed to by B<in>.
+The encrypted data should be stored in B<out> and the amount of data written to
+B<*outl> which should not exceed B<outsize> bytes.
+OP_cipher_update() may be called multiple times for a single cipher operation.
+It is the responsibility of the cipher implementation to handle input lengths
+that are not multiples of the block length.
+In such cases a cipher implementation will typically cache partial blocks of
+input data until a complete block is obtained.
+B<out> may be the same location as B<in> but it will not partially overlap.
```
Comment (paulidale) [inline doc/man7/provider-cipher.pod]: _should not_ instead of _will not_ ?

File: doc/man7/provider-cipher.pod [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+The same expectations apply to B<outsize> as documented for
+L<EVP_EncryptUpdate(3)> and L<EVP_DecryptUpdate(3)>.
+
+OP_cipher_final() completes an encryption or decryption started through previous
+OP_cipher_encrypt_init() or OP_cipher_decrypto_init(), and OP_cipher_update()
```
Comment (paulidale) [inline doc/man7/provider-cipher.pod]: `OP_cipher_decrypt_init`

**Comment by mattcaswell** [comment]
Fixup commit pushed addressing all nits above.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Comment by bernd-edlinger** [comment]
Hmm. that is funny because github does seem to be updated as it used to be.
@levitte

**Comment by bernd-edlinger** [comment]
I mean it is not updated, also #9470 did not make it.

**Pull request closed without merging**